ui/ui-components/components/record/SampleRecordCard.tsx

```diff
@@ -46,30 +48,64 @@ export interface SampleRecordCardProps {
   reloadKey?: string;
   warning?: string;
   groupId?: string;
+
+  record?: RecordType;
+  groups?: Models.GroupType[];
+  destinations?: Models.DestinationType[];
 }
 
 // This is exported so we can manipulate it in tests, but is unused elsewhere.
 export const isConfigUI = grouparooUiEdition() === "config";
 
+const cookies = new Cookies();
+
 const getCachedSampleRecordId = (modelId: string): string => {
-  return globalThis.localStorage?.getItem(
-    `sampleRecord:${grouparooUiEdition()}:${modelId}`
-  );
+  if (isBrowser()) {
+    const localStorageRecordId = globalThis.localStorage?.getItem(
+      `sampleRecord:${grouparooUiEdition()}:${modelId}`
+    );
+    if (localStorageRecordId) {
+      setCachedSampleRecordId(modelId, localStorageRecordId);
+      globalThis.localStorage?.removeItem(
+        `sampleRecord:${grouparooUiEdition()}:${modelId}`
+      );
+      return localStorageRecordId;
```

Comment:
Is this codepath just for migrating an existing sample record?  If so, I'm not too worried about folks needing to re-select the same record again.  If you do only need this for a migration, please add a TODO so we remember to remove it in a later release

Reply:
Added a comment and TODO.

I agree the migration may not be 100% necessary. Would you prefer I just remove it while I'm still here?

Reply:
Not that I feel strongly, but but I would vote to remove it - the user experience won't suffer that much, and at the moment, the theme of simplifying branching codepaths on our front-end is something we are focusing on

Reply:
I was on the fence about including it. Removed!